controller/lib/pid/pid.cpp

```diff
@@ -65,11 +64,18 @@ bool PID::Compute() {
   unsigned long now = Hal.millis();
   unsigned long timeChange = (now - lastTime);
   if (timeChange >= SampleTime) {
+    // compute actual samples time-difference to take jitter into account in
```

Comment:
Is this designed for only slight jitter in sample times, or for arbitrary non-uniform time sampled PID? The latter seems to be a pretty complicated matter, e.g. here's a [research paper](https://aip.scitation.org/doi/pdf/10.1063/1.5066491) about it.

In general, is there a reference for the math changes you're making in this PR?

Reply:
No reference, memories and notes from my controls class at ERAU... you are right, this design is supposed to be for small jitter only, but I have no safeguard in case the jitter is "too big"

Reply:
Amendment on this previous answer: it should work properly with a big jitter, problem is that the PID gains might be out of tune when the jitter is too big.